sonar-plugin-api/src/main/java/org/sonar/api/issue/filter/FilterableIssue.java

```diff
@@ -17,20 +17,30 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.api.issue;
+package org.sonar.api.issue.filter;
 
-import org.sonar.api.ExtensionPoint;
-import org.sonar.api.batch.BatchSide;
+import org.sonar.api.rule.RuleKey;
+
+import java.util.Date;
 
 /**
- * @since 3.6
- * @deprecated since 4.0 use {@link org.sonar.api.issue.batch.IssueFilter}
+ * @since 5.3
  */
-@Deprecated
-@BatchSide
-@ExtensionPoint
-public interface IssueFilter {
+public interface FilterableIssue {
+
+  public String componentKey();
+
+  public RuleKey ruleKey();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) "public" is redundant in this context. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2333)